libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/SalesforceTLSSocketFactory.java

```diff
@@ -48,8 +48,16 @@
  */
 public class SalesforceTLSSocketFactory extends SSLSocketFactory {
 
+    private static SalesforceTLSSocketFactory INSTANCE;
     private SSLSocketFactory ssLSocketFactory;
 
+    public static SalesforceTLSSocketFactory getInstance() throws KeyManagementException, NoSuchAlgorithmException {
+        if (INSTANCE == null) {
+            INSTANCE = new SalesforceTLSSocketFactory();
+        }
+        return INSTANCE;
+    }
+
     public SalesforceTLSSocketFactory() throws KeyManagementException, NoSuchAlgorithmException {
```

Comment:
@ashishmody I deliberately didn't make this ```private```. There might be some use cases where I want a new instance as opposed to the same.